### PR TITLE
SQL related improvements

### PR DESCRIPTION
**Pull request by srknzl**
### Changes

* If connection is not alive, client's uuid is used instead of remote uuid. fixes that. Note: Due to this change I think we should backport this to 4.2.x 

* Changes isRowSet logic to make it faster

* Doc or API doc changes

* Changes 'should reject iteration after close()' test to run close() at the middle of iteration with page size of 1. It was flaky in the previous version because if the result is received before close() is called, it will fail. Also adds catch() to rejected fetchPromise to suppress UnhandledPromiseRejectionWarning  Note: Also due to this change I think we should backport this to 4.2.x
* ignore empty modules from api doc

### CODE REVIEW

File: src/sql/SqlPage.ts [hunk 1]
```diff
@@ -53,6 +54,7 @@ export class SqlPage {
 
     /**
      * This is needed for better mocking. Constructor mocking is not trivial with sinon.
+     * @internal
```
Comment (mdumandag) [inline src/sql/SqlPage.ts:57]: Do we need this, since the class is marked as internal?

File: src/sql/SqlResult.ts [hunk 2]
```diff
@@ -211,7 +212,7 @@ export class SqlResultImpl implements SqlResult {
 
     isRowSet(): Promise<boolean> {
         return this.executeDeferred.promise.then(() => {
-            return this.updateCount === Long.fromInt(-1);
+            return this.rowMetadata !== null;
```
Comment (mdumandag) [inline src/sql/SqlResult.ts:215]: just to be sure, `rowMetadata` can't be `undefined`, right?

File: src/sql/SqlResult.ts [hunk 3]
```diff
@@ -242,6 +243,7 @@ export class SqlResultImpl implements SqlResult {
         // Prevent ongoing/future fetch requests
         if (!this.fetchDeferred) {
             this.fetchDeferred = deferredPromise<SqlPage>();
+            this.fetchDeferred.promise.catch(()=>{});
```
Comment (mdumandag) [inline src/sql/SqlResult.ts:246]:
```suggestion
            this.fetchDeferred.promise.catch(() => {});
```

File: src/sql/SqlResult.ts [hunk 4]
```diff
@@ -255,7 +257,11 @@ export class SqlResultImpl implements SqlResult {
         return this.closeDeferred.promise;
     }
 
-    /** Called when next page of the result is received. */
+    /**
+     * Called when next page of the result is received.
+     * @param page
+     * @internal
```
Comment (mdumandag) [inline src/sql/SqlResult.ts:263]: Do we need to mark private functions as internal?

File: src/sql/SqlResult.ts [hunk 5]
```diff
@@ -282,6 +292,7 @@ export class SqlResultImpl implements SqlResult {
     /**
      * Used by {@link next}.
      * @returns the current row.
+     * @internal
```
Comment (mdumandag) [inline src/sql/SqlResult.ts:295]: Do we need to mark private functions as internal?

File: src/sql/SqlResult.ts [hunk 6]
```diff
@@ -372,6 +389,7 @@ export class SqlResultImpl implements SqlResult {
     /**
      * Used by {@link next}.
      * @returns if there are rows to be iterated.
+     * @internal
```
Comment (mdumandag) [inline src/sql/SqlResult.ts:392]: Do we need to mark private functions as internal?

File: src/sql/SqlService.ts [hunk 7]
```diff
@@ -378,11 +380,24 @@ export class SqlServiceImpl implements SqlService {
         return this.executeStatement(sqlStatement);
     }
 
+    /**
+     * Sends a close request on a connection for an SQL result using its query id.
+     * @param connection The connection the request will be sent to
+     * @param queryId The query id that defines the SQL result
+     * @internal
```
Comment (mdumandag) [inline src/sql/SqlService.ts:387]: I see that this class is internal, do we need to mark methods as internal again?

File: src/sql/SqlPage.ts (hunk 1 above)
Comment (srknzl) [reply src/sql/SqlPage.ts:57]: 👍🏻  no we don't

File: src/sql/SqlResult.ts (hunk 2 above)
Comment (srknzl) [reply src/sql/SqlResult.ts:215]: no it cant be

File: src/sql/SqlResult.ts (hunk 4 above)
Comment (srknzl) [reply src/sql/SqlResult.ts:263]: also the class is internal anyway, going to delete it

Comment (srknzl) [reply src/sql/SqlResult.ts:263]: I am going to delete @internal labels of all methods in an internal class

File: src/sql/SqlResult.ts [hunk 8]
```diff
@@ -350,7 +362,6 @@ export class SqlResultImpl implements SqlResult {
 
     /**
      * Checks if there are rows to iterate in a recursive manner, similar to a non-blocking while block
-     * @internal
```
Comment (srknzl) [inline src/sql/SqlResult.ts:353]: I guess we had this before, therefore I added other internal labels. Let's delete this one too

File: src/sql/SqlService.ts [hunk 9]
```diff
@@ -253,15 +251,15 @@ export class SqlServiceImpl implements SqlService {
 
     /**
      * Converts an error to HazelcastSqlException and returns it. Used by execute, close and fetch
-     * @internal
```
Comment (srknzl) [inline src/sql/SqlService.ts:256]: Same here we had them before, these led me to add internal labels to other methods

**Review by mdumandag — APPROVED** [review]

**Comment by srknzl** [comment]
verify

**Pull request merged**